Automattic/wp-calypso
# Add smooth animation to FoldableCard component

#### Proposed Changes
Currently, `FoldableCard` component makes content disappear immediately. To have a more smooth option we are adding several parameters to enable smooth animation. This will prevent `content` from `display: none` and instead will set `max-height` property to 0 with transition animation. Also, `contentExpandedStyle` and `contentCollapsedStyle` can help regulate the animation, direction, etc.

P.S. There's probably a room for deeper update of this component or even FoldableCardV2 to have the best performance and look, but we still can keep this version updated in case someone would want to update existing elements.

#### Testing Instructions
- Clone the branch locally or use the Live Links provided below.
- In calypso.localhost and jetpack.cloud.localhost if you are aware of any foldableCard usage please check them to work as before.
- Navigate to `calypso.localhost:3000/devdocs` (or Live link `/devdocs`) and in UI components section find FoldableCard component. You can verify that different versions works correctly.
- On new pricing page`jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1` enable mobile portrait simulator and open a LightBox by clicking `More about ...` button and observe the behavior of Foldable components

**Before**

https://user-images.githubusercontent.com/60262784/190656831-63907227-270c-4553-a2b5-5715cb99a701.mov

**After**

https://user-images.githubusercontent.com/60262784/190656891-d0af3037-0a6f-4b88-a31f-5cc0f907e02e.mov





#### Pre-merge Checklist






Related to #
